cli/command/service/opts.go

```diff
@@ -353,22 +353,24 @@ func (c *credentialSpecOpt) Value() *swarm.CredentialSpec {
 	return c.value
 }
 
-func convertNetworks(ctx context.Context, apiClient client.NetworkAPIClient, networks opts.NetworkOpt) ([]swarm.NetworkAttachmentConfig, error) {
+func resolveNetworkID(ctx context.Context, apiClient client.NetworkAPIClient, networkIDOrName string) (string, error) {
+	nw, err := apiClient.NetworkInspect(ctx, networkIDOrName, types.NetworkInspectOptions{Scope: "swarm"})
+	if err != nil {
+		return "", err
+	}
+	return nw.ID, nil
```

Comment:
nit: the if is unnecessary since `NetworkIsnpect`  returns a struct not a pointer. It can be replaced by 

```go
return nw.ID, err
```

Reply:
Oh, nice one! I'll add that change

Reply:
done :)